storage/google_drive/drive_client_wrapper.py

```diff
@@ -0,0 +1,172 @@
+
+    log.info('Getting children of folder with id "{}"...'.format(folder_id))
+    page_count = 1
+    while True:
+        log.info('Getting children of folder with id "{}" - got page {}'.format(folder_id, page_count))
+        response = drive_root_folder.list(
+            q="'{}' in parents".format(folder_id),
+            spaces='drive',
+            fields='nextPageToken, files(id, name, mimeType)',
+            pageToken=page_token).execute()
+        for file in response.get('files', []):
+            children.append(file)
+        page_token = response.get('nextPageToken', None)
+        if page_token is None:
+            break
+    log.info('Getting children of folder with id "{}" - done. {} children'.format(folder_id, len(children)))
+    return children
+
+
+def get_children(folder_path):
+    """Returns a list with map elements with the following structure:
+    {
+        'name': ''
+        'id': '',
+        'mimeType': '',
+    }
+    """
+    folder_id = get_path_id(folder_path)
+    return get_children_by_id(folder_id)
+
+
+def get_folder_id(name, parent_id):
+    log.info('Getting id of folder "{}" under parent with id "{}"...'.format(name, parent_id))
+    response = drive_root_folder.list(
+        q="name='{}' and '{}' in parents and mimeType='application/vnd.google-apps.folder'".format(name, parent_id),
+        spaces='drive',
+        fields='files(id)').execute()
+    files = response.get('files', [])
+    if (len(files) == 0):
+        log.error('Folder "{}" not found under parent with id {}.'.format(name, parent_id))
+        exit(1)
+    if (len(files) > 1):
+        log.error('Multiple folders with name "{}" found under parent with id {}.'.format(name, parent_id))
+        exit(1)
+    log.info('Getting id of folder "{}" under parent with id "{}" - done. Folder id is "{}"'.format(name, parent_id, files[0].get('id')))
+    return files[0].get('id')
+
+def get_path_id(path):
+    folders = _split_path(path)
+    folder_id = get_root_id()
+    for folder in folders:
+        folder_id = get_folder_id(folder, folder_id)
+    return folder_id
+
+def _split_path(path):
+    folders = []
+    while path != "":
+        path, folder = os.path.split(path)
+        if folder != "":
+            folders.append(folder)
+    folders.reverse()
+    return folders
+
+def update_file(source_file_path, target_file_id):
+    media = MediaFileUpload(source_file_path,
+                            resumable=True)
+
+    log.info('Updating file with ID "{}" with source file "{}"...'.format(target_file_id, source_file_path))
+    file = drive_root_folder.update(fileId=target_file_id,
+                                    media_body=media,
+                                    fields='name').execute()
+
+    log.info('Updating file with ID "{}" with source file "{}" - done. File name was "{}"'.format(target_file_id, source_file_path, file.get('name')))
+
+def create_file(source_file_path, target_folder_path, target_file_name=None):
```

Comment:
- Please add a comment for behaviour if file already exists (duplicates)
- Consider s/create/add

Reply:
Please make this private